tests/test_tensor/test_linear_tp.py

```diff
@@ -0,0 +1,110 @@
+import torch
+from colossalai.context.parallel_mode import ParallelMode
+from colossalai.tensor import ColoTensor
+from copy import deepcopy
+
+from functools import partial
+
+import colossalai
+import pytest
+import torch
+import torch.multiprocessing as mp
+from colossalai.logging import get_dist_logger
+from colossalai.testing import parameterize, rerun_if_address_is_in_use
+from colossalai.utils.cuda import get_current_device
+from colossalai.utils import free_port
+from colossalai.core import global_context as gpc
+import torch.distributed as dist
+
+def check_equal(A, B):
+    assert torch.allclose(A, B, rtol=1e-3, atol=1e-1) == True
+
+def run_linear_tp1d_row_test():
+    device = get_current_device()
+    dtype = torch.float32
+    DEPTH = 4
+    in_features = 4
+    out_features = 5
+
+    i = gpc.get_local_rank(ParallelMode.PARALLEL_1D)
+
+    layer = torch.nn.Linear(in_features, out_features)
+
+    A_shape = (2, in_features)
+    A_master = torch.randn(A_shape, dtype=dtype, device=device)
+    dist.broadcast(A_master, src=0)
+    A = A_master.clone()
+    A.requires_grad = True
+
+    W_shape = (out_features, in_features)
+    W_master = torch.randn(W_shape, dtype=dtype, device=device)
+    dist.broadcast(W_master, src=0)
+    W = torch.chunk(W_master, DEPTH, dim=-1)[i]
+    W = W.clone()
+    W.requires_grad = True
```

Comment:
The code snippet makes each process has the same tensor.
You can move these to a function.
The code is a little bit redundant

Reply:
Done.